### PR TITLE
Gift cards settings - new setting automatically_fulfill_non_shippable_gift_card

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds new order setting - `automatically_fulfill_non_shippable_gift_card`.

⚠️ This is a PR to feature branch.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3895 Update orderSettingsUpdate mutation](https://app.clickup.com/t/2549495/SALEOR-3895)

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/shop/mutations.py [hunk 1]
```diff
@@ -298,6 +298,11 @@ class OrderSettingsUpdateInput(graphene.InputObjectType):
         "will be marked as unconfirmed. When enabled orders from checkout will "
         "become unfulfilled immediately.",
     )
+    automatically_fulfill_non_shippable_gift_card = graphene.Boolean(
+        required=True,
+        description="When disabled, all new non shippable gift cards "
+        "will require manual fulfillment.",
```
Comment (IKarbowiak) [inline saleor/graphql/shop/mutations.py:304]: I am not sure if the description is good, as when is disabled all gift cards order will require manual fulfillment. Maybe we should describe it from the other side, like:
`When enabled, all non-shippable gift card orders will be fulfilled automatically.`.

File: saleor/graphql/shop/mutations.py [hunk 2]
```diff
@@ -298,6 +298,11 @@ class OrderSettingsUpdateInput(graphene.InputObjectType):
         "will be marked as unconfirmed. When enabled orders from checkout will "
         "become unfulfilled immediately.",
     )
+    automatically_fulfill_non_shippable_gift_card = graphene.Boolean(
+        required=True,
```
Comment (fowczarek) [inline saleor/graphql/shop/mutations.py:302]: Not sure that inputs should be required?

Comment (IKarbowiak) [inline saleor/graphql/shop/mutations.py:302]: IMO both fields should be optional.

Comment (fowczarek) [reply saleor/graphql/shop/mutations.py:302]: @maarcingebala @IKarbowiak What do you think?

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**